### PR TITLE
rpi5-image: Add kernel modules

### PR DESCRIPTION
**Pull request by dsemenets**
Add kernel modules to linux rootfs.
Add dependencies for do_image routine

### CODE REVIEW

File: recipes-core/image/rpi5-image-minimal-domd.bb [hunk 1]
```diff
@@ -5,10 +5,11 @@ require recipes-core/images/core-image-minimal.bb
 # Enable package manager
 EXTRA_IMAGE_FEATURES += "package-management"
 
-RDEPENDS += "rpi-bootfiles"
+DEPENDS:append = " rpi-bootfiles"
```
Comment (lorc) [inline recipes-core/image/rpi5-image-minimal-domd.bb]: This change has nothing to do with kernel modules. Please move it to a separate commit

**Review by lorc — COMMENTED** [review]
Could you please describe what is the reason? Why are you adding all kernel modules? AFAIK, Yocto provides per-module packages like "kernel-module-e1000" so you can install only required modules.

Which modules you need?

**Comment by GrygiriiS** [comment]
> Could you please describe what is the reason? Why are you adding all kernel modules? AFAIK, Yocto provides per-module packages like "kernel-module-e1000" so you can install only required modules.
> 
> Which modules you need?

It's my request - I hit absence of gntdev/gntalloc.

We need default set of modules provided by RPI5 Kernel build for now. And actually in general, no goal and neither requirements to optimize or minimize RPI5 Kernel build.

By default "minimal" image comes without (or with limited number) modules, and it's not suitable to build "base" image (which include modules)  as it has way too many other features enabled and build takes much more time.

**Comment by GrygiriiS** [comment]
@dsemenets Pls, fix comment related to rpi-bootfiles: drop it or add as separate patch then
LGTM `Reviewed-by: Grygorii Strashko <grygorii_strashko@epam.com>`

**Comment by lorc** [comment]
> By default "minimal" image comes without (or with limited number) modules

Well, this is the definition of "minimal" image. 

Actually, proper solution is to introduce `image-xt`  with the product-specific set of modules.

**Comment by GrygiriiS** [comment]
> > By default "minimal" image comes without (or with limited number) modules
> 
> Well, this is the definition of "minimal" image.
> 
> Actually, proper solution is to introduce `image-xt` with the product-specific set of modules.

Huh. Here I do not know. it's already "rpi5-image-minimal-domd.bb". if you think name has to be changed - I'm not against it. But I'd very appreciated if you request will be detailed and specific- like "i_think_its_name_should_be_this" :)

File: recipes-core/image/rpi5-image-minimal-domd.bb (hunk 1 above)
Comment (dsemenets) [reply recipes-core/image/rpi5-image-minimal-domd.bb]: Deleted

**Review by lorc — COMMENTED** [review]
I am fine with the first change. But could you please elaborate your description for the second patch? What was the root cause? How your change fixes it? Or is it just a workaround/experiment? In this case it should be marked as workaround/experiment

**Comment by lorc** [comment]
Also, one more thing: should this change be in this repo? Or in meta-xt-prod-devel-rpi5?

Do we really need all RPI5-based products to include all kernel modules? @GrygiriiS, what is your opinion?

**Comment by oleksiimoisieiev** [comment]
> Also, one more thing: should this change be in this repo? Or in meta-xt-prod-devel-rpi5?
> 
> Do we really need all RPI5-based products to include all kernel modules? @GrygiriiS, what is your opinion?

May I intrude? I think that adding kernel-modules is the common yocto way. Because module configuration depends from the kernel config and can be easily controlled by modifying config. If we use an approach when we're producing the set of kernel modules and add them explicitly then we should change 2 places instead of just one.

**Comment by GrygiriiS** [comment]
> I am fine with the first change. But could you please elaborate your description for the second patch? What was the root cause? How your change fixes it? Or is it just a workaround/experiment? In this case it should be marked as workaround/experiment

@DS Please drop the second patch from this PR as it is not related to it.

**Comment by dsemenets** [comment]
Removed patch related to bootfiles

**Comment by GrygiriiS** [comment]
> > Also, one more thing: should this change be in this repo? Or in meta-xt-prod-devel-rpi5?
> > Do we really need all RPI5-based products to include all kernel modules? @GrygiriiS, what is your opinion?
> 
> May I intrude? I think that adding kernel-modules is the common yocto way. Because module configuration depends from the kernel config and can be easily controlled by modifying config. If we use an approach when we're producing the set of kernel modules and add them explicitly then we should change 2 places instead of just one.

I agree here with @oleksiimoisieiev - modules (at least modules from Kernel repo) are part of the Kernel and first of all depends and enabled by defconfig. So no need to play with module's recipes.
